code/game/objects/items/devices/scanners.dm

```diff
@@ -673,11 +673,20 @@ NANITE SCANNER
 	var/list/discovered = list() //hit a dna console to update the scanners database
 
 /obj/item/sequence_scanner/attack(mob/living/M, mob/living/carbon/human/user)
-	user.visible_message("<span class='notice'>[user] has analyzed [M]'s genetic sequence.</span>")
+	if (!M.has_trait(TRAIT_RADIMMUNE) && !M.has_trait(TRAIT_BADDNA)) //no scanning if its a husk or DNA-less Species
+		user.visible_message("<span class='notice'>[user] has analyzed [M]'s genetic sequence.</span>")
 
-	add_fingerprint(user)
+		add_fingerprint(user)
+
+		gene_scan(M, user, src)
+	else
+		to_chat(user, "<span class='warning'>[M] has no readable genetic sequence!</span>")

```

Comment:
visible_message has a self_message argument, so I think you can merge this into the visible message below.

Reply:
good to know, thanks!